src/rbd.cc

```diff
@@ -1368,6 +1368,56 @@ static void update_snap_name(char *imgname, char **snap)
     *snap = s;
 }
 
+static bool is_valid_name(const char *name)
+{
+   char ch;
+   unsigned int i; 
+
+   if(name == NULL)
+     return true;
+
+   if(strlen(name) == 0)
+     return false;
+   
+   for (i = 0; i < strlen(name); i++ ) {
+      ch = name[i];
+      if ( !( ( ch >= '0' && ch <= '9' ) ||
```

Comment:
There is no restriction of anything to ascii-only in ceph afaik. I don't see much advantage to doing that. Anything passing any kind of user input to the shell should be escaping it properly. Also, the shell inteprets things that might cause problems like that before getting to ceph. 

Mistakes like passing $#$#* instead of '$#$#*' can't be caught by ceph, since ceph only gets the arguments after shell expansion, so it just sees 00*.

There are downsides to restricting to ascii-only:
- existing tools built using rbd or ceph commands with non-ascii characters may break
- users can't name things in non-ascii-supported languages
- migrating everywhere names are used to have the same restrictions is a lot of work

So overall I don't think restricting the character set for names is very beneficial, and has non-trivial costs.